src/base/NekRSProblemBase.C

```diff
@@ -629,23 +636,39 @@ NekRSProblemBase::externalSolve()
   if (_t_step <= 1000)
     nekrs::verboseInfo(true);
 
-  // Tell NekRS what the time step size is
-  nekrs::initStep(_timestepper->nondimensionalDT(step_start_time),
-                  _timestepper->nondimensionalDT(_dt),
-                  _t_step);
-
   // Run a nekRS time step. After the time step, this also calls UDF_ExecuteStep,
   // evaluated at (step_end_time, _t_step) == (nek_step_start_time + nek_dt, t_step)
-  int corrector = 1;
-  bool converged = false;
-  do
+  if (_fp_iteration)
   {
-    converged = nekrs::runStep(corrector++);
-  } while (!converged);
-
-  // TODO: time is somehow corrected here
-  nekrs::finishStep();
+    const PostprocessorValue * iter = &getPostprocessorValueByName("fp_iteration");
+    std::cout << "Current fixed point iteration number read in NekRSProblemBase is " << *iter
+              << std::endl; // JUST FOR TESTING
+    if (*iter == 1)
+    {
+      nekrs::initStep(_timestepper->nondimensionalDT(step_start_time),
+                      _timestepper->nondimensionalDT(_dt),
+                      _t_step);
+    }
+    bool converged = false;
+    converged = nekrs::runStep(*iter);
 
+    nekrs::finishStep();
+  }
+  else
+  {
+    // Tell NekRS what the time step size is
+    nekrs::initStep(_timestepper->nondimensionalDT(step_start_time),
+                    _timestepper->nondimensionalDT(_dt),
+                    _t_step);
+    int corrector = 1;
+    bool converged = false;
+    do
+    {
+      converged = nekrs::runStep(corrector++);
+    } while (!converged);
+    // TODO: time is somehow corrected here
```

Comment:
Has `TODO` item been accomplished i.e. the time correction? @regerdavid

Reply:
I don't believe that that is my TODO comment (looks like it exists in the current master branch of cardinal) so I'm not sure if that is resolved or not.